tests/tel2puml/find_unique_graphs/test_otel_data_holder.py

```diff
@@ -0,0 +1,184 @@
+
+from unittest.mock import patch
+from sqlalchemy import text
+
+from tel2puml.find_unique_graphs.otel_ingestion.otel_data_model import (
+    OTelEvent,
+    NodeModel,
+    SQLDataHolderConfig,
+)
+from tel2puml.find_unique_graphs.otel_ingestion.otel_data_holder import (
+    SQLDataHolder,
+)
+
+
+class TestSQLDataHolder:
+    """Tests for class SQLDataHolder."""
+
+    @staticmethod
+    def test_sql_data_holder_init(
+        mock_sql_config: SQLDataHolderConfig,
+    ) -> None:
+        """Tests the init method."""
+
+        holder = SQLDataHolder(mock_sql_config)
+        assert holder.batch_size == 10
+        assert holder.node_models_to_save == []
+        assert holder.node_relationships_to_save == []
+        with patch.object(holder, "create_db_tables") as mock_create_db_table:
+            holder.create_db_tables()
+            mock_create_db_table.assert_called_once()
+
+    @staticmethod
+    def test_create_db_tables(mock_sql_config: SQLDataHolderConfig) -> None:
+        """Tests the create_db_tables method."""
+
+        holder = SQLDataHolder(mock_sql_config)
+        with patch.object(
+            holder.base.metadata, "create_all"
+        ) as mock_create_all:
+            holder.create_db_tables()
+            mock_create_all.assert_called_once_with(holder.engine)
+
+    @staticmethod
+    def test_convert_otel_event_to_node_model(
+        mock_sql_config: SQLDataHolderConfig, mock_otel_event: OTelEvent
+    ) -> None:
+        """Tests convert_otel_event_to_node_model method."""
+
+        holder = SQLDataHolder(mock_sql_config)
+        node_model = holder.convert_otel_event_to_node_model(mock_otel_event)
+        assert isinstance(node_model, NodeModel)
+        assert node_model.job_name == "test_job"
+        assert node_model.job_id == "123"
+        assert node_model.event_type == "test_event"
+        assert node_model.event_id == "456"
+        assert node_model.start_timestamp == 1723544154817793024
+        assert node_model.end_timestamp == 1723544154817993024
+        assert node_model.application_name == "test_app"
+
+    @staticmethod
+    def test_add_node_relations(
+        mock_sql_config: SQLDataHolderConfig, mock_otel_event: OTelEvent
+    ) -> None:
+        """Tests add_node_relations method."""
+
+        holder = SQLDataHolder(mock_sql_config)
+        holder.add_node_relations(mock_otel_event)
+        assert len(holder.node_relationships_to_save) == 2
+        assert holder.node_relationships_to_save[0]["parent_id"] == "456"
+        assert holder.node_relationships_to_save[0]["child_id"] == "101"
+        assert holder.node_relationships_to_save[1]["parent_id"] == "456"
+        assert holder.node_relationships_to_save[1]["child_id"] == "102"
+
+    @staticmethod
+    def test_save_data(
+        mock_sql_config: SQLDataHolderConfig, mock_otel_event: OTelEvent
+    ) -> None:
+        """Tests save_data method"""
+
+        holder = SQLDataHolder(mock_sql_config)
+        with patch.object(
```

Comment:
Again would like to see that the data is replicated correctly into the relevant tables, not just mocking and asserting the call